### PR TITLE
Run3-Sim101E Remove warnings in a few classes of SimTracker/SiPixelDigitizer

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Remove warnings in a few classes of SimTracker/SiPixelDigitizer

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimTracker/SiPixelDigitizer/test/PixelDigisTest.cc [hunk 1]
```diff
@@ -103,7 +103,7 @@ using namespace std;
 // class decleration
 //
 
-class PixelDigisTest : public edm::EDAnalyzer {
+class PixelDigisTest : public edm::one::EDAnalyzer<> {
```
Comment (makortel) [inline SimTracker/SiPixelDigitizer/test/PixelDigisTest.cc:106]: This class appears to need
```suggestion
class PixelDigisTest : public edm::one::EDAnalyzer<edm::one::SharedResources> {
```
and
```c++
  usesResource(TFileService::kSharedResource);
```
in the constructor.

File: SimTracker/SiPixelDigitizer/test/PixelSimHitsTest.cc [hunk 2]
```diff
@@ -70,7 +70,7 @@ using namespace edm;
 
 //#define CHECK_GEOM
 
-class PixelSimHitsTest : public edm::EDAnalyzer {
+class PixelSimHitsTest : public edm::one::EDAnalyzer<> {
```
Comment (makortel) [inline SimTracker/SiPixelDigitizer/test/PixelSimHitsTest.cc:73]: Same here

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

core suggested modifications are implemented

**Comment by perrotta** [comment]
+1

**Pull request merged**